src/Components/Homepage.js

```diff
@@ -1,38 +1,58 @@
-import React, {Component} from 'react';
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+
 import Header from './Header';
 import Main from './Main';
 import Footer from './Footer';
 import ErrorBoundary from './ErrorBoundary';
-import './Homepage.css';
 import { MovieModal } from "./MovieModal";
+import { MovieInfo } from "./MovieInfo";
 
-class Homepage extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isAddMovieModalShown: false
-        }
-    }
+import './Homepage.css';
+
+export const Homepage = () => {
+    const useToggle = (isComponentShown = false) => {
+        const [flag, setFlag] = useState(isComponentShown);
 
-    handleToggleAddMovieModal = () => {
-        this.setState(prevState => ({
-            isAddMovieModalShown: !prevState.isAddMovieModalShown
-        }))
+        const toggle = useCallback(() => {
+            setFlag(!flag)
```

Comment:
it's better to use function when you are related on previous state
```js
setFlag(prev => !prev)
```

Reply:
fixed